test/generic_subscription_wrapper.cc

```diff
@@ -0,0 +1,65 @@
+#include "napi.h"
+
+#include <thread>
+#include <stdexcept>
+#include <cstdint>
+#include <future>
+#include <utility>
+#include <functional>
+#include <atomic>
+
+#if (NAPI_VERSION > 3)
```

Comment:
I think this needs to be >4 as the threadsafe functions were introduced in N-API version 4 -> https://nodejs.org/api/n-api.html#n_api_napi_threadsafe_function

Reply:
`>= 4` would probably be clearer, because it mentions the N-API version needed for the tsfn while at the same time indicating with which minimum version of N-API it works.

Reply:
I agreed >=4 would be better. I just realised that it was ok because it was a > and was coming back to leave that comment, but @gabrielschulhof beat me to it:)